File: 1341-split-a-string-in-balanced-strings/split-a-string-in-balanced-strings.py
class Solution:
    def balancedStringSplit(self, s: str) -> int:

        lcount = 0
        rcount = 0
        result = 0

        for c in s:
            if c == 'L':
                lcount += 1
            else:
                rcount += 1
                
            if (lcount == rcount):
                result += 1
        
        return result

File: 1341-split-a-string-in-balanced-strings/split-a-string-in-balanced-strings.java
class Solution {
    public int balancedStringSplit(String s) {
        int lCount = 0, rCount = 0, result = 0;

        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == 'L') {
                lCount++;
            } else {
                rCount++;
            }

            if (lCount == rCount) {
                result++;
            }
        }

        return result;
    }
}